src/misc/bleeding.ts

```diff
@@ -0,0 +1,204 @@
+import { ParticleBuilder } from 'com.destroystokyo.paper';
+import { Location, Material, Particle } from 'org.bukkit';
+import { BlockFace } from 'org.bukkit.block';
+import { Waterlogged } from 'org.bukkit.block.data';
+import { Entity, EntityType, LivingEntity } from 'org.bukkit.entity';
+import { EntityDamageEvent } from 'org.bukkit.event.entity';
+import { DamageCause } from 'org.bukkit.event.entity.EntityDamageEvent';
+
+/**
+ * Adds a bleeding effect for damageable entities by randomly generating blood puddles and splashes around entity
+ *
+ * Currently implemented features:
+ *
+ * - Damage amount and health determine how much the entity bleeds
+ * - Injured entities bleed more violently when damaged again
+ * - Bleeding exponentially decreases if not injured
+ * - Armor, clothing etc... will affect bleeding amount
+ * - It is now possible to determine entity health just by looking how much they bleed
+ *
+ * @author Juffel
```

Comment:
@Laetta Thoughts on author tags? I think they're pointless because Git has the same information available.

Reply:
Yeah, Git already provides that information